mesheryctl/pkg/utils/helpers.go

```diff
@@ -693,3 +693,52 @@ func IsAdapterValid(manifestArr []Manifest, adapterManifest string) bool {
 
 	return false
 }
+
+// GetCurrentContext returns the current context name and context struct
+func GetCurrentContext(tempCtxName string) (string, config.Context, error) {
+	// if the user has mentioned a temporary context in the -c flag, change the context and proceed to reset
+	mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+	if err != nil {
+		return "", config.Context{}, errors.Wrap(err, "error processing config")
+	}
+
+	if tempCtxName == "" {
+		return mctlCfg.CurrentContext, mctlCfg.Contexts[mctlCfg.CurrentContext], nil
+	}
+	if tempCtxtStruct, val := mctlCfg.Contexts[tempCtxName]; val {
+		return tempCtxName, tempCtxtStruct, nil
+	}
+	// if the user specifies a context that is not in the config.yaml file, throw an error and show the available contexts
+	log.Errorf("\n\"%s\" context does not exist. The available contexts are:", tempCtxName)
+	for context := range mctlCfg.Contexts {
+		log.Errorf("%s", context)
+	}
+	return "", config.Context{}, errors.Errorf("\n\"%s\" context does not exist", tempCtxName)
+}
+
+// DownloadDockerComposeFile fetches docker-compose.yaml based on passed context if it does not exists
+func DownloadDockerComposeFile(ctx config.Context, force bool) (string, error) {
+	if _, err := os.Stat(DockerComposeFile); os.IsNotExist(err) || force {
+		fileURL := ""
+
+		if ctx.Channel == "edge" {
+			fileURL = "https://raw.githubusercontent.com/layer5io/meshery/master/docker-compose.yaml"
+			ctx.Version = "edge"
```

Comment:
Should the version also be "edge"? Or should it be "latest"? The channel and version being "edge" feels a bit off. What does everyone suggest?

Reply:
Yea this might raise an issue.. Currently the versions are edge-latest and stable-latest. so the version should either be set as latest or be leftout. first alternative is better.

Reply:
Also you dont need a GetCurrentContext function... The mesheryctl config class already has that inplace. A good place to look into is config.go

Reply:
if a change is required, feel free to edit config.go instead

Reply:
check out the updated code, I don't think we should get rid of GetCurrentContext. If we remove it, we will need to add duplicate code at many places

Reply:
@aneeshnema Yes, that would be best